std/ndn/security.go

```diff
@@ -18,11 +18,18 @@ type Signature interface {
 	SigValue() []byte
 }
 
-// Signer is the interface of the signer of a packet.
+// Signer is the interface of a NDN packet signer.
 type Signer interface {
-	SigInfo() (*SigConfig, error)
+	// SigInfo returns the configuration of the signature.
+	Type() SigType
+	// KeyName returns the key name of the signer.
+	KeyName() enc.Name
+	// EstimateSize gives the approximate size of the signature in bytes.
 	EstimateSize() uint
-	ComputeSigValue(enc.Wire) ([]byte, error)
+	// Sign computes the signature value of a wire.
+	Sign(enc.Wire) ([]byte, error)
+	// Public returns the public key of the signer or nil.
+	Public() ([]byte, error)
```

Comment:
I think you need to separate `Signer` with `PrivateKey`. `Signer` executes the signing action with a given key, its fields are related to the signing action (sequence number, timestamp, nonce, etc.), instead of the key itself.

Reply:
Hmm on starting the refactor, I realised this won't fit the SHA256 signer (since there is no key). Then we'd need either another name / implementation for SHA256 signer, or a separate signer implementation (want to avoid this complexity).

Reply:
> > sequence number, timestamp, nonce, etc.
> 
> Trying to eliminate these if possible. We shouldn't be using signed interest that can be replayed. SVSv3 is the first to move away.

I don't follow. SeqNum/Nonce/Timestamp were designed to prevent replay attacks.

Reply:
Yeah, the intention is to eventually eliminate signed interest.

For now I don't plan to support any of these optional fields in NDNd.

Reply:
understood